sycl/include/sycl/ext/oneapi/matrix/matrix-unified.hpp

```diff
@@ -59,21 +59,9 @@ class wi_data {
               joint_matrix<Grp, Type, UseJm, NumRows, NumCols, LayoutJm> &);
 
 public:
-  size_t length() {
-#if defined(__NVPTX__)
-    return jm.cuda_impl.wi_marray.size();
-#else
-    return __spirv_JointMatrixWorkItemLengthINTEL(jm.spvm);
-#endif
-  };
+  size_t length() { return jm.cuda_impl.wi_marray.size(); };
```

Comment:
I think you need to keep if defined NVPTX
if the else: generate an error that users should use the intel namespace. But not sure whether we can generate static error like this.
@yubingex007-a11y, what do you suggest in this case?

Reply:
Added runtime_error